### ISSUE TITLE
SOSL find clause WITH USER_MODE or SYSTEM_MODE

### ISSUE THREAD

**Issue by adangel** [apex-dev-tools/apex-parser#42]
Reposting the issue from https://github.com/google/summit-ast/issues/53 here:

apex-parser doesn't support `WITH USER_MODE` and `WITH SYSTEM_MODE` when used in SOSL.

The Apex documentation is not clear, because:
- According to https://developer.salesforce.com/docs/atlas.en-us.248.0.apexcode.meta/apexcode/apex_classes_enforce_usermode.htm and https://developer.salesforce.com/docs/atlas.en-us.248.0.apexcode.meta/apexcode/apex_classes_with_security_enforced.htm it should be available for both SOQL and SOSL. The examples are all SOQL, though...
  > You can indicate the mode of the operation by using WITH USER_MODE or WITH SYSTEM_MODE in your SOQL or SOSL query. 
- The SOSL documentation doesn't list this: https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_sosl_syntax.htm

Is the following code valid Apex code?

```apex
public inherited sharing class soqlExample {

  public static String example(){
    String SecondarySearchList = 'test';

    List<List<SObject>> accountList = [
      FIND :SecondarySearchList
      IN NAME FIELDS
      RETURNING Account(Id, Account.Name WHERE ID = '' LIMIT 100)
      WITH USER_MODE
    ];
  }
}
```

If yes, then someone needs to update the SOSL documentation and we probably need to enhance the parser:
https://github.com/apex-dev-tools/apex-parser/blob/2f9160c4a33e5acf1a9014f0b67de0496df0ff47/antlr/ApexParser.g4#L826-L838

If no, then someone just needs to clarify the Apex documentation and remove the SOSL reference.

**Comment by kjonescertinia** [comment]
That code can be deployed so think this is an issue that needs fixing.

**Comment by pwrightcertinia** [comment]
Fix now available in v4.1.0 release, thanks for the report!
